### PR TITLE
Add ReflectionHelper for properties

### PR DESCRIPTION
**Pull request by TheMulhima**

### CODE REVIEW

**Comment by fifty-six** [comment]
95% of the reason for the helper in the first place is avoiding .GetValue/.SetValue as they're really slow and getting private fields is relatively common due to necessity when modding. Is there a point to this besides it being kinda convenient? It'll end up being slower than caching the PropertyInfo(s) as static fields yourself which you should be doing anyways. Without emitting IL there's not really a huge reason to have it

**Comment by TheMulhima** [comment]
The only real reason it's kinda convient. If you don't want it in mapi I'll move it to satchel or something

**Comment by fifty-six** [comment]
I don't care a *ton* but if you're gonna put it in the API you should go for IL, should just be a call instruction anyways. If properties are common it's fine either way - I don't need more fractured libraries. If you don't want to bother I'll just merge it as-is though

File: Assembly-CSharp/ReflectionHelper.cs [hunk 1]
```diff
@@ -357,5 +403,108 @@ private static Delegate CreateSetStaticFieldDelegate<TField>(FieldInfo fi)
...
+        /// <summary>
+        ///     Set a property on an object using a string.
+        /// </summary>
+        /// <param name="obj">Object/Object of type which the field is on</param>
+        /// <param name="name">Name of the field</param>
+        /// <param name="value">Value to set the field to</param>
+        /// <typeparam name="TProperty">Type of property</typeparam>
+        /// <typeparam name="TObject">Type of object being passed in</typeparam>
+        [PublicAPI]
+        public static void SetPropertySafe<TObject, TProperty>(TObject obj, string name, TProperty value)
```
Comment (fifty-six) [inline Assembly-CSharp/ReflectionHelper.cs:467]: Same deal with this one

**Comment by TheMulhima** [comment]
I would love to use IL to make it faster but I don't know how that works unfortunately

**Comment by TheMulhima** [comment]
I would love to use IL to make it faster but I don't know how that works unfortunately

**Comment by TheMulhima** [comment]
I'll give it a try today and see

**Comment by TheMulhima** [comment]
Whoops. didn't realise renaming branch would close pr
 I'll make a new pr ig

**Pull request closed without merging**